src/Gelf/Transport/KeepAliveRetryTransportWrapper.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Gelf\Transport;
+
+use Gelf\MessageInterface as Message;
+
+class KeepAliveRetryTransportWrapper extends AbstractTransport
+{
+    /**
+     * @const string
+     */
+    const NO_RESPONSE = "Graylog-Server didn't answer properly, expected 'HTTP/1.x 202 Accepted', response is ''";
+
+    /**
+     * @var HttpTransport
+     */
+    protected $transport;
+
+    /**
+     * KeepAliveRetryTransportWrapper constructor.
+     *
+     * @param HttpTransport $transport
+     */
+    public function __construct(HttpTransport $transport)
```

Comment:
I propose a few changes:

- Make `$transport` a generic `TransportInterface`
- `int $maxRetries = 2` - Allow to specify a max number of retries. 0 means infinite retries
- `?callable $exceptionMatcher = null` - The optional matcher should be `callable(Throwable):bool`. THe returned boolean will tell whether the exception can be retried. If no matcher is passed into the class, all exceptions are retried. This allows you to extract the http/keepalive specific logic.